app/views/users/show.html.erb

```diff
@@ -0,0 +1,54 @@
+<div class=" container row mx-auto">
+<div class="col-8 mt-5 mx-auto shadow p-3 mb-5 bg-white rounded row">
+
+  <div class="container">
+
+    <h3 class="fw-bold">User Events</h3>
+    <div class="row">
+      <% @user.created_events.each do |event| %>
+        <div class="col-6">
+        <%= render partial: "layouts/event", locals: {event: event} %>
+        </div>
+      <%end%>
+    </div>
+  </div>
+  <hr>
+  <div class="container mt-3">
+  <h3 class="fw-bold">Attending</h3>
+  <div class="row mt-2">
+
+    <div class="col-6">
+
+      <div>
+        <h3>Previously Attended Events</h3>
+        <%= display_past_events(User.find(session[:current_user_id])) %>
+      </div>
+
+    </div>
+
+    <div class="col-6">
+
+      <div>
+        <h3>Upcoming Events</h3>
+        <%@user.attended_events.each do |event| %>
+          <%= display_upcoming_events(@user.id, event) %>
+        <%end%>
+      </div>
```

Comment:
- [ ] There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.
Inside your User Controller, you can have something like `
@created_events = current_user.events` then on the view you do `@created_events do |event| ...`
You can create a helper method in your application controller, `current_user = User.find(session[:current_user_id])` and use it anywhere in your application.

Reply:
Replaced the data queries from users/show.html.erb and moved them to the show action of Users controller